cmd/temporalite/ui.go

```diff
@@ -6,29 +6,45 @@
 
 package main
 
+// This file should be the only one to import ui-server packages.
+// This is to avoid embedding the UI's static assets in the binary when the `headless` build tag is enabled.
 import (
-	// This file should be the only one to import ui-server packages.
-	// This is to avoid embedding the UI's static assets in the binary when the `headless` build tag is enabled.
+	"strings"
+
+	provider "github.com/temporalio/ui-server/v2/plugins/fs_config_provider"
 	uiserver "github.com/temporalio/ui-server/v2/server"
 	uiconfig "github.com/temporalio/ui-server/v2/server/config"
 	uiserveroptions "github.com/temporalio/ui-server/v2/server/server_options"
 
 	"github.com/temporalio/temporalite"
 )
 
-func newUIOption(frontendAddr string, uiIP string, uiPort int) temporalite.ServerOption {
-	return temporalite.WithUI(uiserver.NewServer(uiserveroptions.WithConfigProvider(newUIConfig(
+func newUIOption(frontendAddr string, uiIP string, uiPort int, configDir string) (temporalite.ServerOption, error) {
+	cfg, err := newUIConfig(
 		frontendAddr,
 		uiIP,
 		uiPort,
-	))))
+		configDir,
+	)
+	if err != nil {
+		return nil, err
+	}
+	return temporalite.WithUI(uiserver.NewServer(uiserveroptions.WithConfigProvider(cfg))), nil
 }
 
-func newUIConfig(frontendAddr string, uiIP string, uiPort int) *uiconfig.Config {
-	return &uiconfig.Config{
-		TemporalGRPCAddress: frontendAddr,
-		Host:                uiIP,
-		Port:                uiPort,
-		EnableUI:            true,
+func newUIConfig(frontendAddr string, uiIP string, uiPort int, configDir string) (*uiconfig.Config, error) {
+	cfg := &uiconfig.Config{
+		Host: uiIP,
+		Port: uiPort,
+	}
+	if configDir != "" {
+		if err := provider.Load(configDir, cfg, "temporalite-ui"); err != nil {
+			if !strings.HasPrefix(err.Error(), "no config files found") {
```

Comment:
Hrmm, would have liked to see if we could prevent the error instead of baking in expectations around its English message. I forget the config file loading whether it's easy to know what files to check for existence eagerly. But I guess it's not _that_ bad since you have a unit test that confirms this conditional works.

Reply:
Ideally it would be great if the Load function returned an error we can check for with `errors.Is` since that is the best way of avoiding tedious string comparisons. I also wanted to avoid having to duplicate the file resolution algorithm applied by `Load`. I reckon for now the unit test check will at least blow up if the algorithm changes or the error changes.